sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/memory.scala

```diff
@@ -202,6 +202,8 @@ class MemorySink(val schema: StructType, outputMode: OutputMode) extends Sink wi
             s"Output mode $outputMode is not supported by MemorySink")
       }
     } else {
+      // TODO: THIS IS NOT RIGHT!
+      data.collect()
```

Comment:
this added line is to demonstrate the related issue
this should really be reverted